packages/api/src/command/medical/document/process-doc-query-webhook.ts

```diff
@@ -15,37 +15,86 @@ const { log } = out(`Doc Query Webhook`);
 const isSandbox = Config.isSandbox();
 const DOWNLOAD_WEBHOOK_TYPE = "medical.document-download";
 const CONVERSION_WEBHOOK_TYPE = "medical.document-conversion";
+type WebhookType = typeof DOWNLOAD_WEBHOOK_TYPE | typeof CONVERSION_WEBHOOK_TYPE;
+
+export const handleWebhookBeingSent = async ({
+  patient,
+  requestId,
+}: {
+  patient: Patient;
+  requestId: string;
+}) => {
+  const downloadStatus = patient.data?.documentQueryProgress?.download?.status;
+  const convertStatus = patient.data?.documentQueryProgress?.convert?.status;
+
+  const isDownloadFinished = downloadStatus === "completed" || downloadStatus === "failed";
+  const isConvertFinished = convertStatus === "completed" || convertStatus === "failed";
+
+  if (isDownloadFinished) {
+    await processDocQueryProgressWebhook({
+      patient,
+      requestId,
+      progressType: "download",
+      status: downloadStatus,
+    });
+  } else if (isConvertFinished) {
+    await processDocQueryProgressWebhook({
+      patient,
+      requestId,
+      progressType: "convert",
+      status: convertStatus,
+    });
+  }
```

Comment:
Can't we have both at the same time?

If we call `handleWebhookBeingSent` when download is finished, then call it again when conversion is finished, it seems `isDownloadFinished` will be truthy both times, leading to 2 `download` WHs, no?

Reply:
Updated this logic lmk if it makes more sense

Reply:
Taken to [Slack](https://metriport.slack.com/archives/C04CXLSAF7V/p1707163404592099?thread_ts=1706650131.915859&cid=C04CXLSAF7V)